src/main/java/racing/Car.java

```diff
@@ -3,18 +3,30 @@
 public class Car {
     private static final char sign = '-';
     private int allMeter = 0;
+    private Display display;
 
-    public int run(int meter){
-        allMeter = allMeter + meter;
-        for(int i = 0 ; i < allMeter ; i++){
-            System.out.print(sign);
+    public Car(Display display) {
+        this.display = display;
+    }
+
+    public int run(int inputNumber) {
+        allMeter = allMeter + returnMeterByCondition(inputNumber);
+        for (int i = 0; i < allMeter; i++) {
+            display.print(sign);
         }
         stopRunning();
         return allMeter;
     }
+    protected int returnMeterByCondition(int inputNumber){
+        int moveableNumber = 4;
+        if (inputNumber < moveableNumber) {
+            return 0;
+        }
+        return inputNumber;
+    }
```

Comment:
랜덤값을 외부에서 주입받도록 하여 자동차 스스로가 움직일 수 있는지 없는지 판단하는 설계 👍 
대신 입력받은 숫자만큼 이동거리가 증가하는게 아니라 한 칸만 증가하는 미션입니다!
미션요구사항을 참고하시면 좋을 것 같아요!
```suggestion
    protected int returnMeterByCondition(int inputNumber){
        int moveableNumber = 4;
        if (inputNumber < moveableNumber) {
            return 0;
        }
        
        this.allMeter = this.allMeter + 1;
        return this.allMeter;
    }
```